salt/utils/openstack/nova.py

```diff
@@ -192,6 +192,21 @@ def get_entry_multi(dict_, pairs, raise_error=True):
     return {}
 
 
+def get_endpoint_url_v3(catalog, service_type, region_name):
+    endpoints = [
+        endpoint_entry['url']
+        for service_entry in catalog
+        if service_entry['type'] == service_type
+        for endpoint_entry in service_entry['endpoints']
+        if endpoint_entry['region'] == region_name
+        if endpoint_entry['interface'] == 'public'
```

Comment:
can this be changed into an actual for loop so that we can read it?

I know that this format is executed faster, but it is very difficult to read.

Reply:
Done